packages/trace-viewer/src/ui/uiModeView.tsx

```diff
@@ -83,62 +79,129 @@ export const UIModeView: React.FC<{}> = ({
     const guid = new URLSearchParams(window.location.search).get('ws');
     const wsURL = new URL(`../${guid}`, window.location.toString());
     wsURL.protocol = (window.location.protocol === 'https:' ? 'wss:' : 'ws:');
-    const connection = new TestServerConnection(wsURL.toString());
-    setTestServerConnection(connection);
-    setIsLoading(true);
-    setWatchedTreeIds({ value: new Set() });
-    updateRootSuite(baseFullConfig, new TeleSuite('', 'root'), [], undefined);
-    (async () => {
-      const status = await connection.runGlobalSetup();
-      if (status === 'passed')
-        await refreshRootSuite(connection);
-      setIsLoading(false);
-      const { hasBrowsers } = await connection.checkBrowsers();
-      setHasBrowsers(hasBrowsers);
-    })();
+    setTestServerConnection(new TestServerConnection(wsURL.toString()));
   }, []);
 
+  // Load tests on startup.
+  React.useEffect(() => {
+    inputRef.current?.focus();
+    setIsLoading(true);
```

Comment:
Shouldn't `reloadTests()` set loading to true itself, so that all other callers have it automatically? Or perhaps we already do this in "the main routine", so can just remove from here?

Reply:
This is to give it a more "immediate" feedback.